pyanaconda/ui/gui/spokes/subscription.py

```diff
@@ -0,0 +1,750 @@
+# in the authentication mode combo box
+class AuthenticationMethod(IntEnum):
+    USERNAME_PASSWORD = 0
+    ORG_KEY = 1
+
+
+class SubscriptionSpoke(NormalSpoke):
+    """Subscription spoke provides the Connect to Red Hat screen."""
+    builderObjects = ["subscription_window"]
+
+    mainWidgetName = "subscription_window"
+    uiFile = "spokes/subscription.glade"
+    help_id = "SubscriptionSpoke"
+
+    category = SystemCategory
+
+    icon = "application-certificate-symbolic"
+    title = CN_("GUI|Spoke", "_Connect to Red Hat")
+
+    # main notebook pages
+    REGISTRATION_PAGE = 0
+    SUBSCRIPTION_STATUS_PAGE = 1
+
+    @classmethod
+    def should_run(cls, environment, data):
+        """The Subscription spoke should run only if the Subscription module is available."""
+        return is_module_available(SUBSCRIPTION)
+
+    def __init__(self, *args):
+        NormalSpoke.__init__(self, *args)
+
+        # connect to the Subscription DBus module API
+        self._subscription_module = SUBSCRIPTION.get_proxy()
+
+        # connect to the Network DBus module API
+        self._network_module = NETWORK.get_proxy()
+
+        # get initial subscription request from the Subscription module
+        self._subscription_request = self._get_subscription_request()
+
+        self._authentication_method = AuthenticationMethod.USERNAME_PASSWORD
+
+        self._registration_error = ""
+        self._registration_phase = None
+        self._registration_controls_enabled = True
+
+    # common spoke properties
+
+    @property
+    def ready(self):
+        """The subscription spoke is always ready."""
+        return True
+
+    @property
+    def status(self):
+        # TODO: status message depends on registration/unregistration handling
+        # - shows registration phases when registration + subscription is ongoing
+        # - otherwise shows not-registered/registered/error
+        return ""
+
+    @property
+    def mandatory(self):
+        """The subscription spoke is mandatory if Red Hat CDN is set as installation source."""
+        # TODO: depends on source switching, mandatory if Red Hat CDN is set as installation source
+        return True
+
+    @property
+    def completed(self):
+        return self.subscription_attached
+
+    @property
+    def sensitive(self):
+        # the Subscription spoke should be always accessible
+        return True
+
+    # common spoke methods
+
+    def apply(self):
+        log.debug("Subscription GUI: apply() running")
+        # Set System Purpose data to the Subscription DBus module,
+        # it makes sense to send the rest only right before a
+        # registration attempt.
+        self._set_system_purpose_data_to_subscription_module()
+
+    def refresh(self):
+        log.debug("Subscription GUI: refresh() running")
+        # update subscription request from the Subscription module
+        self._subscription_request = self._get_subscription_request()
```

Comment:
This method should set up the UI elements from the current request.

Reply:
Added a content refresh method called from here.